src/App.js

```diff
@@ -151,115 +66,74 @@ const App = () => {
-        }).searchClient
-      )
-
-      setMSClient(
-        new Meilisearch({
-          host: baseUrl,
-          apiKey,
-          clientAgents,
-        })
-      )
-    }
+  useEffect(() => {
+    setInstantMeilisearchClient(
+      instantMeilisearch(baseUrl, apiKey, {
+        primaryKey: 'id',
+        clientAgents,
+      }).searchClient
+    )
+
+    setMeilisearchJsClient(
+      new Meilisearch({
+        host: baseUrl,
+        apiKey,
+        clientAgents,
+      })
+    )
   }, [apiKey])
 
-  // Check if a modal asking for API Key should be displayed
-  React.useEffect(() => {
-    const shouldDisplayModal = async () => {
-      try {
-        await MSClient.getIndexes()
-      } catch (err) {
-        console.log(err)
-        dialog.show()
-      }
+  useEffect(async () => {
+    const isInstanceRunning = await meilisearchJsClient.isHealthy()
+    setIsMeilisearchRunning(isInstanceRunning)
+    if (isInstanceRunning) {
+      setRequireApiKeyToWork(await hasAnApiKeySet())
+      dialog.setVisible(await shouldDisplayApiKeyModal(meilisearchJsClient))
+      getIndexesList()
     }
-    if (requireApiKeyToWork) shouldDisplayModal()
-  }, [requireApiKeyToWork])
-
-  // Get the list of indexes
-  React.useEffect(() => {
-    getIndexesList()
-  }, [MSClient, currentIndex?.uid])
+  }, [meilisearchJsClient])
 
   return (
     <ApiKeyContext.Provider value={{ apiKey, setApiKey }}>
       <Wrapper>
-        <InstantSearch
-          indexName={currentIndex ? currentIndex.uid : ''}
-          searchClient={ISClient}
-        >
-          {isApiKeyBannerVisible && (
-            <ApiKeyAwarenessBanner onClose={handleBannerClose} />
-          )}
-          {showCloudBanner && <CloudBanner />}
-          <Header
-            indexes={indexes}
+        {isApiKeyBannerVisible && (
+          <ApiKeyAwarenessBanner
+            onClose={() => setIsApiKeyBannerVisible(false)}
+          />
+        )}
+        {showCloudBanner && <CloudBanner />}
+        {isMeilisearchRunning ? (
+          <Body
             currentIndex={currentIndex}
+            indexes={indexes}
             setCurrentIndex={setCurrentIndex}
             requireApiKeyToWork={requireApiKeyToWork}
-            client={MSClient}
-            refreshIndexes={getIndexesList}
-            isBannerVisible={isApiKeyBannerVisible}
+            getIndexesList={getIndexesList}
+            isApiKeyBannerVisible={isApiKeyBannerVisible}
           />
-          <Body>
-            {/* <Sidebar /> */}
-            <Box
-              width={928}
-              m="0 auto"
-              py={4}
-              display="flex"
-              flexDirection="column"
-            >
-              {isMeilisearchRunning ? (
-                <Content currentIndex={currentIndex} />
-              ) : (
-                <NoMeilisearchRunning />
-              )}
-            </Box>
-          </Body>
-        </InstantSearch>
+        ) : (
+          <NoMeilisearchRunning />
```

Comment:
Do you really need this? Isn't it handled already by the `<EmptyView />`?

Reply:
Their role are different. The `EmptyView` is for when your meilisearch instance is running but you have no documents in your index, whereas the `NoMeilisearchRunning` is for when you have no meilisearch instance running